plugin.video.BlanketFort/default.py

```diff
@@ -0,0 +1,515 @@
+                else:
+                    addLink(url[0],name.encode('utf-8', 'ignore'),thumbnail,fanArt,desc,genre,date,True)
+            except:
+                print 'There was a problem adding link - '+name.encode('utf-8', 'ignore')
+
+
+def get_params():
+        param=[]
+        paramstring=sys.argv[2]
+        if len(paramstring)>=2:
+            params=sys.argv[2]
+            cleanedparams=params.replace('?','')
+            if (params[len(params)-1]=='/'):
+                params=params[0:len(params)-2]
+            pairsofparams=cleanedparams.split('&')
+            param={}
+            for i in range(len(pairsofparams)):
+                splitparams={}
+                splitparams=pairsofparams[i].split('=')
+                if (len(splitparams))==2:
+                    param[splitparams[0]]=splitparams[1]
+        return param
+
+
+def getFavorites():
+        for i in json.loads(open(favorites).read()):
+            name = i[0]
+            url = i[1]
+            iconimage = i[2]
+            try:
+                fanArt = i[3]
+                if fanArt == None:
+                    raise
+            except:
+                if addon.getSetting('use_thumb') == "true":
+                    fanArt = iconimage
+                else:
+                    fanArt = fanart
+            addLink(url,name,iconimage,fanArt,'','','')
+
+
+def addFavorite(name,url,iconimage,fanart):
+        favList = []
+        if os.path.exists(favorites)==False:
+            print 'Making Favorites File'
+            favList.append((name,url,iconimage,fanart))
+            a = open(favorites, "w")
+            a.write(json.dumps(favList))
+            a.close()
+        else:
+            print 'Appending Favorites'
+            a = open(favorites).read()
+            data = json.loads(a)
+            data.append((name,url,iconimage,fanart))
+            b = open(favorites, "w")
+            b.write(json.dumps(data))
+            b.close()
+
+
+def rmFavorite(name):
+        a = open(favorites).read()
+        data = json.loads(a)
+        for index in range(len(data)):
+            if data[index][0]==name:
+                del data[index]
+                b = open(favorites, "w")
+                b.write(json.dumps(data))
+                b.close()
+                return
+
+
+def play_playlist(name, list):
+        playlist = xbmc.PlayList(1)
+        playlist.clear()
+        item = 0
+        for i in list:
+            item += 1
+            info = xbmcgui.ListItem('%s) %s' %(str(item),name))
+            playlist.add(i, info)
+        xbmc.executebuiltin('playlist.playoffset(video,0)')
+
+
+def addDir(name,url,mode,iconimage,fanart,description,genre,date,showcontext=True):
+        u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)+"&fanart="+urllib.quote_plus(fanart)
+        ok=True
+        liz=xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)

```

Comment:
Please remove the iconImage and thumbnailImage inits, they are deprecated. Use .setArt() instead.